src/main/java/org/miracum/etl/fhirtoomop/mapper/ObservationMapper.java

```diff
@@ -1100,23 +1100,25 @@ private void setValueQuantityInObservation(
       Quantity valueQuantity,
       String observationId) {
     basisObservation.setValueAsNumber(valueQuantity.getValue());
-    var quantityUnitCodingFormat =
-        new Coding().setCode(valueQuantity.getCode()).setSystem(valueQuantity.getSystem());
-    var valueQuantityUnitConcept =
-        findOmopConcepts.getConcepts(
-            quantityUnitCodingFormat,
-            effectiveDateTime.toLocalDate(),
-            bulkload,
-            dbMappings,
-            observationId);
-    if (valueQuantityUnitConcept != null) {
-
-      basisObservation.setUnitConceptId(valueQuantityUnitConcept.getConceptId());
-      basisObservation.setUnitSourceValue(
-          valueQuantity.getUnit() == null
-              ? valueQuantityUnitConcept.getConceptCode()
-              : valueQuantity.getUnit());
-      addToList(observations, basisObservation);
+    if (valueQuantity.hasSystem()){
```

Comment:
Ok let me check this.

Reply:
This check is required because later we are using the system to get some code on line 1105. And if the system does not exist and this check is not there then the ETL pipeline will fail.